test/integration/integration_test.cc

```diff
@@ -283,6 +283,39 @@ TEST_P(IntegrationTest, HittingGrpcFilterLimitBufferingHeaders) {
               HeaderValueOf(Headers::get().GrpcStatus, "2")); // Unknown gRPC error
 }
 
+TEST_P(IntegrationTest, TrailersWithHttp1AndHttp2) {
+  setUpstreamProtocol(FakeHttpConnection::Type::HTTP2);
+  Http::TestHeaderMapImpl response_trailers{{"response1", "trailer1"}, {"response2", "trailer2"}};
+
+  initialize();
+  codec_client_ = makeHttpConnection(lookupPort("http"));
+  auto encoder_decoder =
+      codec_client_->startRequest(Http::TestHeaderMapImpl{{":method", "POST"},
+                                                          {":path", "/test/long/url"},
+                                                          {":scheme", "http"},
+                                                          {":authority", "host"},
+                                                          {"TE", "trailers"},
+                                                          {"Accept", "*/*"}});
+  request_encoder_ = &encoder_decoder.first;
+  codec_client_->sendData(*request_encoder_, 0, true);
+
+  auto response = std::move(encoder_decoder.second);
```

Comment:
to reproduce seeing the trails with curl

1. remove these lines , 308 and the EXPECT_EQ statements below.
2. bazel run  test/integration:integration_test --test_arg="-l trace" --test_filter="*TrailersWithHttp1*"
3. curl -vv --raw localhost:$envoyport -H "TE: trailers"  --data-binary @/tmp/request.data --output -

envoyport will be available in the test output.

Reply:
However if I use a standard tcp client it does show the trailers

```
  auto tcp_client = makeTcpConnection(lookupPort("http"));
  tcp_client->write("POST / HTTP/1.1\r\nHost: host\r\nTE: trailers\r\nAccept: */*\r\n\r\n");

  waitForNextUpstreamRequest();
  upstream_request_->encodeHeaders(default_response_headers_, false);
  upstream_request_->encodeData(5, false);
  upstream_request_->encodeTrailers(response_trailers);
  //response->waitForEndStream();

  ASSERT_TRUE(upstream_request_->complete());
  //ASSERT_TRUE(response->complete());

  tcp_client->waitForData("\r\n\r\n", false);
  std::string response = tcp_client->data();
```